docs/development/visualize/development-create-visualization.asciidoc

```diff
@@ -0,0 +1,307 @@
+
+["source","js"]
+-----------
+{
+    name: 'my_new_vis',
+    title: 'My New Vis',
+    icon: 'my_icon',
+    description: 'Cool new chart',
+    editorController: 'default',
+    editorConfig: {
+       optionsTemplate: '<my-custom-options-directive></my-custom-options-directive>' // or
+       optionsTabs: [
+           { title: 'tab 1', template: '<div>....</div> },
+           { title: 'tab 2', template: '<my-custom-options-directive></my-custom-options-directive>' }
+       ]
+    }
+  }
+-----------
+
+[[development-custom-editor]]
+==== custom editor controller
+You can create a custom editor controller. To do so pass an Editor object (the same format as VisController class). 
+You can make your controller take extra configuration which is passed to the editorConfig property.
+
+["source","js"]
+-----------
+import { VisFactoryProvider } from 'ui/vis/vis_factory';
+
+class MyEditorController {
+    constructor(el, vis) {
+      this.el = el;
+      this.vis = vis;
+   }
+   render(visData) {
+      return new Promise(resolve => {
+         ...
+         resolve('done rendering');
+      }),
+   destroy() {
+      console.log('destroying');
+   }
+}
+
+export default function MyNewVisType(Private) {
+  const VisFactory = Private(VisFactoryProvider);
+
+  return VisFactory.createAngularVisualization({
+    name: 'my_new_vis',
+    title: 'My New Vis',
+    icon: 'my_icon',
+    description: 'Cool new chart',
+    editorController: MyEditorController,
+    editorConfig: { my: 'custom config' }
+  });
+}
+-----------
+
+[[development-visualization-request-handlers]]
+=== Visualization Request Handlers
+By default visualizations will use courier request handler. They can also choose to use any of the other provided 
+request handlers. Its also possible to define your own request handler 
+(which you can then register to be used by other visualizations)
+
+[[development-default-request-handler]]
+==== courier request handler
+'courier' is the default request handler which works with out 'default' side bar editor.
+
+[[development-none-request-handler]]
+==== `none` request handler
+Using 'none' as your request handles means your visualization does not require search.
+
+[[development-custom-request-handler]]
+==== custom request handler
+You can define your custom request handler by providing a function with the following definition:
+`function (vis, appState, uiState, searchSource) { ... }`
+
+[[development-visualization-response-handlers]]
+=== Visualization Response Handlers
+Response handler will receive the data from request handler and an instance of Vis object. Its job is to convert the data to a format visualization can use.
+by default 'none' request handler is used which just passes the data it receives back, but can be changed to 'basic' (which VislibVisTypeFactory uses) or to a custom one.
```

Comment:
What exactly will be done with the return value of this function? To what (scope) variable/prop will it be assigned?